src/dimension_slice.c

```diff
@@ -734,8 +734,8 @@ ts_dimension_slice_scan_iterator_get_by_id(ScanIterator *it, int32 slice_id,
 	ts_scan_iterator_start_or_restart_scan(it);
 	ti = ts_scan_iterator_next(it);
 	Assert(ti);
```

Comment:
So Coverity doesn't understand the meaning of this assert? I was thinking of marking this as a false positive...

Reply:
It's because Coverity runs against the release build.